### PR TITLE
Validate data no simulation

### PR DESCRIPTION
**Pull request by ew3361zh**
This PR seeks to address #737 in which a user can utilize the InputManager's data validation capacity without running a simulation afterward.

1. Adds flag `terminate_after_validation` (tav) along with gnu arg for said flag. Default is set to False.
2. Updates logic in main.py `execute_simulations()` to include this flag's value.
3. Dumps nondata pools but doesn't save variables in case of tav. This was tested and variables pool was empty which makes sense given all the variables are added during a simulation.
4. Updates unit testing accordingly.

Notes:
- The default for exclude_info_maps was set to True in master but since the gnu arg flag is to set it to True, I changed the default back to False. This may have been accidentally changed on a local branch and merged into master.
- I also added an option for the user to control the early_termination flag from the gnu args.
- I'm differentiating early_termination and terminate_after_validation. Early_termination ends the validation process if the Input Manager encounters unfixable invalid critical data and sends back a False bool to main.py which will prevent a simulation from being run. Terminate_after_validation is the option to not run a simulation even after a full validation/fixing of the data by Input Manager has been achieved, regardless of the value of the early_termination flag.

### CODE REVIEW

File: main.py
Comment (Fjordrunner) [inline main.py]: This is definitely nitpicky but I prefer including a verb in most Boolean names...the first suggestions that come to mind are `is_early_termination` or `terminate_early`. I think that `terminate_early` is better here since it matches the following variable, `terminate_after_validation`. This applies to everywhere else where this variable is used, including the test file in case we do go through with this.

File: main.py
Comment (Fjordrunner) [inline main.py]: Maybe `early_termination` isn't actually the best variable name here since it doesn't shine light on termination occurring early because of invalid critical data?

**Review by Fjordrunner — CHANGES_REQUESTED** [review]
Overall, this looks really good: thanks for the manageable PR size! My comments are somewhat nitpicky but the more I read into the new code, the less I liked that variable name, happy to have a discussion on it though.

**Review by Isgotkowitz — APPROVED** [review]
LGTM.

**Comment by Fjordrunner** [comment]
@ew3361zh @PooyaHekmati I'm not the biggest fan of the variable name "early_termination," as I highlight above. The most striking issue here is that early termination doesn't have anything to do with users themselves terminating runs early. I feel as if the name suggests that a user is manually exiting or aborting a run via an early termination, as opposed to there being missing critical data. If this Boolean variable is related to there being invalid critical data, I suggest a different name entirely with the focus being on the absent critical data. I propose the name "invalid_data_found" or "invalid_critical_data_found," because it can be reasonably implied that if such data is found, the program would terminate early.

**Comment by ew3361zh** [comment]
> @ew3361zh @PooyaHekmati I'm not the biggest fan of the variable name "early_termination," as I highlight above. The most striking issue here is that early termination doesn't have anything to do with users themselves terminating runs early. I feel as if the name suggests that a user is manually exiting or aborting a run via an early termination, as opposed to there being missing critical data. If this Boolean variable is related to there being invalid critical data, I suggest a different name entirely with the focus being on the absent critical data. I propose the name "invalid_data_found" or "invalid_critical_data_found," because it can be reasonably implied that if such data is found, the program would terminate early.

What about something like `terminate_early_on_invalid_data`?

**Review by PooyaHekmati — CHANGES_REQUESTED** [review]
I agree that `early_termination` is not a good choice as it is 1) ambiguous, and 2) similar to `eager_termination` that is used in input manager. You can rename the GNU arg to something like `only_run_validation` or something of that nature. Users care about what the code does, not what it doesn't. 
Also, if this arg is true, then we won't run the simulation. The question is, why do we need to call `execute_simulations()` when we are not going to run the simulation? This doesn't seem right.

**Comment by ew3361zh** [comment]
Updates:

- [x] `early_termination` renamed to `eager_termination` to match its use elsewhere and for clarity.
- [x] `eager_termination `removed from gnu arguments - not something users should be changing in gnu args.
- [x] `terminate_after_validation` -> `only_run_validation`
- [x] `only_run_validation` being True triggers call to new function in main.py `run_validation()` which just runs data validation.
- [x] `execute_simulations()` only called if `only_run_validation` is False.
- [x] Unit testing updated.

**Review by PooyaHekmati — APPROVED** [review]
[LGTM](https://lgtm.lol/i/714)

**Comment by PooyaHekmati** [comment]
The code looks good, all comments have been resolved; and the PR has the needed approvals. Overriding the requested changes to get this PR merged and unblock other PRs that are waiting for this.

**Pull request merged**
